content/advanced/fine-grained-reactivity.mdx

```diff
@@ -0,0 +1,248 @@
+
+setCount(0); // Modifies the count value to 0
+
+console.log(count()) // prints "0"
+```
+
+Here, `count` serves as the getter, and `setCount` functions as the setter.
+
+### Effects
+
+Effects are operations triggered by changes in one or more signals they depend on.
+To create effects in Solid, there is the `createEffect` function.
+
+```js=
+import { createSignal, createEffect } from "solid-js";
+
+const [count, setCount] = createSignal(0);
+
+createEffect(() => {
+  console.log(count());
+});
+```
+
+The effect takes a callback function that is re-invoked whenever any of the signals it relies on, such as `count`, undergoes a change.
+
+## Building a reactive system
+
+To grasp the concept of reactivity, it's often helpful to construct a reactive system from scratch. 
+The following example will follow the observer pattern, where data entities (signals) will maintain a list of their subscribers (effects). When a signal changes, all the subscribers will be notified.
+
+### Initial code structure
+
+Here is a basic code outline to begin:
+
+```jsx
+function createSignal() {}
+
+function createEffect() {}
+
+const [count, setCount] = createSignal(0);
+
+createEffect(() => {
+  console.log("The count is " + count());
+});
+```
+
+### Implementing `createSignal`
+
+The `createSignal` function performs two main tasks:
+1. Initialize the value, in this case `count`, to 0.
+2. Return an array with two elements: a getter and a setter function.
+
+Implementation:
+
+```tsx
+function createSignal(initialValue) {
+  let value = initialValue;
+  
+  function getter() {
+    return value;
+  }
+  
+  function setter(newValue) {
+    value = newValue;
+  }
+  
+  return [getter, setter];
+}
+```
+
+This allows retrieval of the current value via the getter and modification through the setter. 
+However, reactivity is not yet implemented.
+
+### Implementing `createEffect`
+
+The basic form of `createEffect` is simple:
+
+```jsx
+function createEffect(fn) {
+  fn();
+}
+```
+
+The core functionality emerges when linking `createSignal` and `createEffect`.
+
+### Linking signals and effects
+
+In order to introduce reactivity, the following steps are necessaary:
+1. Maintain a reference to the current subscriber function.
+2. Register this function during effect creation.
+3. Add the function to a subscriber list when accessing a signal.
+4. Notify all subscribers upon updating a signal.
+
+This can be shown as:
+
+```jsx
+let currentSubscriber = null;
+
+function createSignal(initialValue) {
+  let value = initialValue;
+  const subscribers = new Set();
+  
+  function getter() {
+    if (currentSubscriber) {
+      subscribers.add(currentSubscriber);
+    }
+    return value;
+  }
+  
+  function setter(newValue) {
+    value = newValue;
+    for (const subscriber of subscribers) {
+      subscriber();
+    }
+  }
+  
+  return [getter, setter];
+}
+
+function createEffect(fn) {
+  currentSubscriber = fn;
+  fn();
+  currentSubscriber = null;
```

Comment:
This line is wrong, after the function is executed the "current subscriber" doesn't disappear, it gets reset to the value that it was before the function executed. Like it has to work with nested effects and nested memos basically.